### PR TITLE
fix JENKINS-47617 CHANGE_BRANCH should return source branch name

### PR DESCRIPTION
**Pull request by fengxx**
CHANGE_BRANCH is correct on bitbucket cloud but not bitbucket server, maybe a typo in origin PR #63  ? cc @zigarn for review

CHANGE_BRANCH is set in branch-api-plugin from getOriginName() 

   ```
   if (head instanceof ChangeRequestSCMHead2) {
      envs.putIfNotNull("CHANGE_BRANCH", ((ChangeRequestSCMHead2) head).getOriginName());
   }
   ```

### CODE REVIEW

**Comment by honnel** [comment]
Works for our bitbucket server v4.13.0 and Jenkins ver. 2.112. When do you plan to release this fix?

**Comment by BrunoLavitForgerock** [comment]
FYI, this PR is not fixing the lightweight checkout on the PRs.
#117 is working fine.

**Comment by honnel** [comment]
We are affected by this issue, too. Any plans for next release?

**Comment by bhipple** [comment]
Getting this release would be great; we're pinned to version `2.2.3` to avoid the `CHANGE_BRANCH` regression, and would like to upgrade, since the newer tagged release functionality looks nice and newer versions of blue ocean require this.

**Comment by benjaminfuchs** [comment]
@BrunoLavitForgerock Can you give more information. I thought this PR would fix also the lightweight checkout?

**Comment by tarioch** [comment]
@amuniz  Anyhting we can do to help getting this merged?

**Comment by dcoraboeuf** [comment]
Hi, we're facing the same issue. Created #133 but closed it in favour of this PR.

**Comment by casz** [comment]
gentle ping @amuniz @rsandell @vivek I know your extremely busy perhaps you should reconsider who has maintainer 😅 this been open for a long time, simple fix and fixes an outstanding issue.

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java [hunk 1]
```diff
@@ -666,7 +666,7 @@ private void retrievePullRequests(final BitbucketSCMSourceRequest request)
                                 repoOwner,
                                 repository,
                                 repositoryType,
-                                branchName,
+                                pull.getSource().getBranch().getName(),
```
Comment (nfalco79) [inline src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java]: Maybe save this to a new variabile near branchName named originalBranchName just to remember in the future what is It and why take from pull instance instead use already avaiable branchName.

**Comment by andreaseger** [comment]
Is anybody working on this?

**Comment by jan-xyz** [comment]
Is there a workaround for this to get the source branch in a declarative pipeline? I doubt that it gets merged any time soon and it breaks our use case to have no access to the CHANGE_BRANCH

**Comment by casz** [comment]
Still waiting on my maintainer permission, I would very much like to see this released

**Comment by smipi1** [comment]
Hi @amuniz, @rsandell, and @vivek,
Any ETA on a release containing this?

**Pull request merged**